File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> ans;
        for(int i = 0; i<nums.size(); i++) {
            int idx = abs(nums[i]) - 1;
            nums[idx] *= -1;
            if(nums[idx] > 0) ans.push_back(abs(nums[i]));
        }

        return ans;
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> ans = new ArrayList<>();
        for(int i = 0; i<nums.length; i++) {
            int idx = Math.abs(nums[i]) - 1;
            nums[idx] *= -1;
            if(nums[idx] > 0) ans.add(Math.abs(nums[i]));
        }

        return ans;
    }
}